src/indexer.c

```diff
@@ -603,6 +603,23 @@ static void hash_partially(git_indexer *idx, const uint8_t *data, size_t size)
 
 static int write_at(git_indexer *idx, const void *data, off64_t offset, size_t size)
 {
+#ifdef NO_MMAP
+	size_t remaining_size = size;
+	const char *ptr = (const char *)data;
+
+	/* Handle data size larger that ssize_t */
+	while (remaining_size > 0) {
+		ssize_t nb;
+		HANDLE_EINTR(nb, p_pwrite(idx->pack->mwf.fd, (void *)ptr,
+					  remaining_size, offset));
+		if (nb <= 0)
+			return -1;
+
+		ptr += nb;
+		offset += nb;
+		remaining_size -= nb;
+	}
```

Comment:
I think that I would have place the return here so that the entire function was encapsulated either in the `#if` or the `#else` blocks.  Not major, but having to jump down to see what was happening next was jarring.  It would have been nice to have each branch fully within their own space so that I didn't have to jump around.

Reply:
I agree. Having self-contained blocks makes it easier to read & you know for certain there is no dependency on anything outside the block.

Reply:
I agree. Having self-contained blocks makes it easier to read & you know for certain there is no dependency on anything outside the block.